temporal-sdk/src/test/java/io/temporal/common/converter/ProtoPayloadConverterTest.java

```diff
@@ -96,6 +102,34 @@ public void testCustomProto() {
     assertEquals(execution, converted);
   }
 
+  @Test
+  public void testProtoMessageType() {
+    DataConverter converter = DataConverter.getDefaultInstance();
+    WorkflowExecution execution =
+        WorkflowExecution.newBuilder()
+            .setWorkflowId(UUID.randomUUID().toString())
+            .setRunId(UUID.randomUUID().toString())
+            .build();
+    Optional<Payloads> data = converter.toPayloads(execution);
+    Payloads payloads = data.get();
+    Object field = payloads.getField(payloads.getDescriptorForType().findFieldByName("payloads"));
+    if (field instanceof List && ((List) field).get(0) instanceof Payload) {
+      Payload payload = (Payload) ((List<?>) field).get(0);
+      Object metadata =
+          payload.getField(payload.getDescriptorForType().findFieldByName("metadata"));
+      if (metadata instanceof List && ((List) metadata).get(1) instanceof MapEntry) {
+        MapEntry<?, ?> secondMetadata = (MapEntry<?, ?>) ((List<?>) metadata).get(1);
+
+        assertEquals("messageType", secondMetadata.getKey());
+        assertEquals(
+            "temporal.api.common.v1.WorkflowExecution",
+            ((ByteString) secondMetadata.getValue()).toString(StandardCharsets.UTF_8));
+        return;
+      }
+    }
```

Comment:
This would be shorter with `@SuppressWarnings("unchecked")` instead of checking `instanceof`—not sure which is preferred.

Reply:
In this test, why not just cast and have permanent expectations instead of put this behind a conditional? Are there cases where this tests follows different code paths in different situations?

Reply:
> Are there cases where this tests follows different code paths in different situations?

No

> In this test, why not just cast and have permanent expectations instead of put this behind a conditional?

When I did that, I got the unchecked casting linting warning. Happy to cast and suppress.

Reply:
I think worth suppressing in the test. What if metadata came back with a different type? You'd succeed the test.